### PR TITLE
4.1: Update UAVCAN for database management options

### PR DESCRIPTION
**Pull request by Hwurzburg**

### CODE REVIEW

**Comment by Hwurzburg** [comment]
@tridge poke!

File: common/source/docs/common-uavcan-setup-advanced.rst [hunk 1]
```diff
@@ -61,7 +61,7 @@ Each UAVCAN ESC or Servo will have an programmed ID or channel address correspon
 There are three parameters that determine which autopilot servo/motor channels are sent to the CAN ESC and/or Servos:
 For the examples below, the values are shown for CAN driver #1.
 
--  :ref:`CAN_D1_UC_NODE<CAN_D1_UC_NODE>` - which is the node ID of the autopilot sending the commands to the ESCs so that there can be differentiation between multiple sources on the CAN bus
+-  :ref:`CAN_D1_UC_NODE<CAN_D1_UC_NODE>` - which is the node ID of the autopilot sending the commands to the ESCs so that there can be differentiation between multiple sources on the CAN bus. When a device with this ID is attached and recognized, it is entered into a database which is stored between power cycles. If multiple devices with the same ID and different hardware IDs (swapping smart batteries, for example, without cycling power) are used, this will require the use of the :ref:`CAN_D1_UC_OPTION<CAN_D1_UC_OPTION>` parameter to allow the database to be reset on the next boot or conflicts in the database to be ignored.
```
Comment (tridge) [inline common/source/docs/common-uavcan-setup-advanced.rst:64]: The CAN_D1_UC_NODE parameter is for the CAN node ID of ArduPilot. The DNA database (dynamic node allocation database) holds node IDs of other nodes, such as all the peripherals. The database stores the node ID and the corresponding hardware identifier.
The UC_OPTION parameter gives methods to either clear the database, or ignore conflicts. A conflict occurs when the same node ID announces itself on the bus with different hardware identifiers.

**Review by tridge — CHANGES_REQUESTED** [review]

File: common/source/docs/common-uavcan-setup-advanced.rst (hunk 1 above)
Comment (Hwurzburg) [reply common/source/docs/common-uavcan-setup-advanced.rst]: think its fixed now...

File: common/source/docs/common-uavcan-setup-advanced.rst [hunk 2]
```diff
@@ -100,6 +100,11 @@ UAVCAN Rangefinder configuration
 
 Set ``RNGFNDx_TYPE`` = 24 to enable UAVCAN rangefinder type. Rangefinder data received over UAVCAN will only be used if the received sensor_id matches the parameter ``RNGFNDx_ADDR``. For AP_Periph firmware based adaptor nodes, this value is 0, so ``RNGFNDx_ADDR`` must be set to 0. Other UAVCAN rangefinders may differ. See also :ref:`UAVCAN Adaptor Node<common-uavcan-adapter-node>` instructions.
 
+UAVCAN Node ID Conflicts
+========================
+
+When a device is attached and recognized, it's node ID and hardware ID are entered into a database which is stored between power cycles. If multiple devices with the same ID and different hardware IDs are used (swapping smart batteries, for example, without cycling power), a conflict will arise in the database. This will require the use of the :ref:`CAN_D1_UC_OPTION<CAN_D1_UC_OPTION>` parameter to allow the database to be reset on the next boot, or conflicts in the database to be ignored.
```
Comment (tridge) [inline common/source/docs/common-uavcan-setup-advanced.rst:106]: change "swapping smart batteries, for example, without cycling power" to "swapping smart batteries with fixed node IDs, for example"

Comment (Hwurzburg) [reply common/source/docs/common-uavcan-setup-advanced.rst]: done

**Pull request merged**